metals/src/main/scala/scala/meta/internal/metals/FallbackMetalsLspService.scala

```diff
@@ -56,26 +59,64 @@ class FallbackMetalsLspService(
   buildServerPromise.success(())
   indexingPromise.success(())
 
+  val pauseables: Pauseable = Pauseable.fromPausables(
+    parseTrees ::
+      compilations.pauseables
+  )
+
   private val files: AtomicReference[Set[AbsolutePath]] = new AtomicReference(
     Set.empty
   )
 
-  override def maybeImportScript(path: AbsolutePath): Option[Future[Unit]] = {
-    if (!path.isScala) None
-    else {
-      val prev = files.getAndUpdate(_ + path)
-      if (prev.contains(path)) None
-      else Some(scalaCli.start(path))
-    }
-  }
-
   override def didClose(params: DidCloseTextDocumentParams): Unit = {
     val path = params.getTextDocument.getUri.toAbsolutePath
     files.getAndUpdate(_ - path)
     super.didClose(params)
     scalaCli.stop(path)
   }
 
+  protected def buildData(): Seq[BuildTool] =
+    scalaCli.lastImportedBuilds.map {
+      case (lastImportedBuild, buildTargetsData) =>
+        Indexer
+          .BuildTool("scala-cli", buildTargetsData, lastImportedBuild)
+    }
+  def buildHasErrors(path: AbsolutePath): Boolean = false
+  protected def check(): Unit = {}
+  protected def didCompileTarget(
+      report: ch.epfl.scala.bsp4j.CompileReport
+  ): Unit = {}
+  def fileWatcher: FileWatcher = NoopFileWatcher
+  def getTargetsInfoForReports(): List[Map[String, String]] = Nil
+
+  def maybeFixAndLoad(
+      path: AbsolutePath,
+      load: () => Future[Unit],
+  ): Future[Unit] =
+    for {
+      _ <-
+        if (!path.isScala) Future.unit
+        else {
+          val prev = files.getAndUpdate(_ + path)
+          if (prev.contains(path)) Future.unit
+          else scalaCli.start(path)
+        }
+      _ <- load()
+    } yield ()
+  protected def onBuildTargetChanges(params: DidChangeBuildTarget): Unit = {
+    compilations.cancel()
+    val scalaCliAffectedServers = params.getChanges.asScala
+      .flatMap { change =>
+        buildTargets.buildServerOf(change.getTarget)
+      }
+      .flatMap(conn => scalaCli.servers.find(_ == conn))
+    importAfterScalaCliChanges(scalaCliAffectedServers)
+  }
+  protected def onInitialized(): Future[Unit] = Future.unit
+  def onMissingSemanticDB(path: AbsolutePath): Unit = {}
+  def scalaCliDirOrFile(path: AbsolutePath): AbsolutePath = path
+  protected val semanticdbs: Semanticdbs = interactiveSemanticdbs
```

Comment:
If you created the semanticdbs as rpeviously in MetalsLangaugeService we can override it here to be just interactive? Or would that be too much trouble?

Reply:
I don't think the override will work for initialisation, since `MetalsLspService` gets created first. I can have both `FileSystemSemanticdbs` and `InteractiveSemanticdbs` in `MetalsLspService` as before to avoid the changes, it's just that `FileSystemSemanticdbs` is not used by the fallback.

Reply:
Ok sure!